src/alexandria3k/data_source.py

```diff
@@ -614,7 +616,30 @@ def __init__(
     def close(self):
         """Terminate use of the data source, freeing associated resources."""
         if self.vdb:
-            self.vdb.close()
+            # Close any active cursor first
+            if hasattr(self, "cursor") and self.cursor:
+                try:
+                    self.cursor.close()
+                except apsw.Error as e:
+                    # Use 'from e' to preserve the exception chain
+                    raise Alexandria3kInternalError(
+                        f"Error closing cursor: {str(e)}"
+                    ) from e
+                self.cursor = None
+
+            # Now close the database
+            try:
+                self.vdb.close()
+            except apsw.BusyError as e:
+                # For specific busy errors
+                raise Alexandria3kInternalError(
+                    f"Database busy error during close: {str(e)}"
+                ) from e
+            except apsw.Error as e:
```

Comment:
This covers the previous one and will be reported through `e`, so I don't see the need to explicitly handle `BusyError`.